api/resource/resource_test.go

```diff
@@ -50,26 +62,38 @@ kind: Deployment
 metadata:
   name: pooh
 `
-	yaml, err := testDeployment.AsYAML()
+	td, err := createTestDeployment()
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("failed to create test deployment: %s", err)
+	}
+	yaml, err := td.AsYAML()
+	if err != nil {
+		t.Fatalf("failed to get yaml: %s", err)
 	}
 	if string(yaml) != expected {
 		t.Fatalf("--- expected\n%s\n--- got\n%s\n", expected, string(yaml))
 	}
 }
 
 func TestResourceString(t *testing.T) {
+	td, err := createTestDeployment()
+	if err != nil {
+		t.Fatal(err)
```

Comment:
Could you add some messages to describe what happens with fatal functions?
Because this test function has multiple points for may happen `Fatal`.